src/server/graphics/default_configuration.cpp

```diff
@@ -370,30 +369,6 @@ auto mir::DefaultServerConfiguration::the_rendering_platforms() ->
             mir::report_exception(error_report);
         }
 
-        rendering_platforms.reserve(rendering_platform_map.size());
-
-        // We want to make the egl-generic platform the last resort
-        // So don't push it into rendering_platforms until the rest are done
-        std::shared_ptr<graphics::RenderingPlatform> egl_generic;
-
-        for (auto const& rp : rendering_platform_map)
-        {
-            if (rp.first != "mir:egl-generic")
-            {
-                rendering_platforms.push_back(rp.second);
-            }
-            else
-            {
-                egl_generic = rp.second;
-            }
-        }
-
-        // If we skipped egl-generic, add it to the end
-        if (egl_generic)
-        {
-            rendering_platforms.push_back(egl_generic);
-        }
-
```

Comment:
I know this code is ugly, but why are we dropping it?

The commit comment "FIXME: Ooops! Don't throw away most of our hardware platforms!" doesn't reflect the change. (It just makes the order non-deterministic.)